cache/models/samples.js

```diff
@@ -47,92 +127,246 @@ function cleanAddAspectToSample(sampleObj, aspectObj, method) {
+ * @param  {Object} opts - Filter options
+ * @returns {Array} - Filtered sample keys array
+ */
+function applyFiltersOnSampKeys(sampKeysArr, opts) {
+  let resArr = sampKeysArr;
+
+  // apply limit and offset if no sort order defined
+  if (!opts.order) {
+    resArr = applyLimitAndOffset(opts, sampKeysArr);
+  }
+
+  // apply wildcard expr on name, if specified
+  if (opts.filter && opts.filter.name) {
+    const filteredKeys = filterByFieldWildCardExpr(
+      resArr, 'name', opts.filter.name
+    );
+    resArr = filteredKeys;
+  }
+
+  return resArr;
+}
+
+/**
+ * Apply field list filter.
+ * @param  {Object} sample - Sample object
+ * @param  {Array} attributes - Sample fields array
+ */
+function applyFieldListFilter(sample, attributes) {
+  // apply field list filter
+  Object.keys(sample).forEach((sampField) => {
+    if (!attributes.includes(sampField)) {
+      delete sample[sampField];
+    }
+  });
+}
+
+/**
+ * Apply filters on sample array list
+ * @param  {Array} sampObjArray - Sample objects array
+ * @param  {Object} opts - Filter options
+ * @returns {Array} - Filtered sample objects array
+ */
+function applyFiltersOnSampObjs(sampObjArray, opts) {
+  let filteredSamples = sampObjArray;
+
+  // apply wildcard expr if other than name because
+  // name filter was applied before redis call
+  if (opts.filter) {
+    const filterOptions = opts.filter;
+    Object.keys(filterOptions).forEach((field) => {
+      if (field !== 'name') {
+        const filteredKeys = filterByFieldWildCardExpr(
+          sampObjArray, field, filterOptions[field]
+        );
+        filteredSamples = filteredKeys;
+      }
+    });
+  }
+
+  // sort and apply limits to samples
+  if (opts.order) {
+    const sortedSamples = sortByOrder(filteredSamples, opts.order);
+    filteredSamples = sortedSamples;
+
+    const slicedSampObjs = applyLimitAndOffset(opts, filteredSamples);
+    filteredSamples = slicedSampObjs;
+  }
+
+  return filteredSamples;
+}
+
 module.exports = {
 
   /**
-   * Retrieves the sample from redis and sends it back in the response.
+   * Retrieves the sample from redis and sends it back in the response. Get
+   * sample and corresponsing aspect from redis and then apply field list
+   * filter is needed. Then attach aspect to sample and return.
+   *
    * @param  {string} sampleName - Sample name
    * @param  {Object} logObject - Log object
    * @param  {String} method - Type of request method
+   * @param  {String} params - Req Query parameters
    * @returns {Promise} - Resolves to a sample object
    */
-  getSampleFromRedis(sampleName, logObject, method) {
-    const aspectName = sampleName.split('|')[ONE];
+  getSampleFromRedis(sampleName, logObject, method, params) {
+    const sampAspArr = sampleName.split('|');
     const commands = [];
 
-    // get sample
+    if (sampAspArr.length < TWO) {
+      throw new redisErrors.ResourceNotFoundError({
+        explanation: 'Incorrect sample name.',
+      });
+    }
+
```

Comment:
The only validation piece I was going to be adding in the API layer is this: 
Check for uniqueness of relatedLinks names within a sample. Do this on sample POST/PATCH/PUT when relatedLinks not empty. Reject with validation error if a relatedLink name is reused within the same sample.